src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -3421,7 +3426,20 @@ int writeFedAuthFeatureRequest(boolean write,
         }
         return totalLen;
     }
+    
+    int writeDataClassificationFeatureRequest (boolean write /* if false just calculates the length */,
+    		TDSWriter tdsWriter) throws SQLServerException {
+        int len = 6; // 1byte = featureID, 4bytes = featureData length, 1 bytes = Version
 
+        if (write) {
+            // Write Feature ID, length of the version# field and Sensitivity Classification Version#
+        	tdsWriter.writeByte((byte) TDS.TDS_FEATUREEXT_DATACLASSIFICATION);
```

Comment:
Let's declare `TDS_FEATUREEXT_DATACLASSIFICATION` as a `byte` instead.

Reply:
This cannot be done, as when this token is received from SQL Server, it is received as `int` as you can see [here](https://github.com/Microsoft/mssql-jdbc/pull/709/files#diff-0a4455f5ffd8e31e9e1badaa7822375eR4067) that is the reason all Feature Ext tokens in the driver are of `int` type.

Reply:
Cannot change it right away, as while reading token, featureId is compared in `int` type, as you can see [here](https://github.com/Microsoft/mssql-jdbc/pull/709/files#diff-0a4455f5ffd8e31e9e1badaa7822375eR4067). If it will be changed for one, it will be changed for others too and need to see its impact on the driver behavior.

Reply:
featureID is a `byte` too, it should not be `int` at the first place.